src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -157,6 +158,93 @@ public boolean isPinned()
         return getTextChannel().clearReactionsById(getId());
     }
 
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(Emote emote, User user)
+    {
+        Checks.notNull(emote, "Emote");
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().getIdLong() == emote.getIdLong())
+            .findFirst().orElse(null);
+        Checks.check(reaction != null,
+            "Cannot remove reaction from a message that doesn't contain the reaction.");
+        return user != null ? reaction.removeReaction(user) : reaction.removeReaction();
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote)
+    {
+        return channel.removeReactionById(getId(), emote);
+    }
+
+    @Nonnull
+    @Override
+    public RestAction<Void> removeReaction(String emote, User user)
+    {
+        Emote e = this.reactions.stream()
+            .map(MessageReaction::getReactionEmote)
+            .filter(r -> r.isEmoji() && r.getEmoji().equals(emote))
+            .map(MessageReaction.ReactionEmote::getEmote)
+            .findFirst().orElse(null);
+        return removeReaction(e, user);
+    }
+
+    @Nonnull
+    @Override
+    public ReactionPaginationAction retrieveReactionUsers(Emote emote)
+    {
+        MessageReaction reaction = this.reactions.stream()
+            .filter(r -> r.getReactionEmote().getEmote().equals(emote))
+            .findFirst().orElse(null);
+        Checks.check(reaction != null,
+            "Cannot retrieve users for a reaction that doesn't exist.");
```

Comment:
Throwing an `IllegalArgumentException` for this is not good. Are you sure that this is true? I would imagine the API allows making this request and returns an empty list. In either case we should allow creating a request even if the cache doesn't line up since the cache might not be accurate on messages anyway.